desc/protoparse/parser.go

```diff
@@ -104,6 +104,14 @@ type Parser struct {
 	// with import statements all being relative to the root of this tree).
 	InferImportPaths bool
 
+	// FileDescriptors is a map of filenames to file descriptor protos, which
```

Comment:
This name is ambiguous. How about `DependencyDescriptors` or `ImportedDescriptors` instead, to be more clear that is only used to resolve dependencies of the files being parsed.

This also needs to be more explicit that this overrides any source files of the same name that might be otherwise loaded via `Accessor`.

Reply:
`ImportedDescriptors` seems reasonable. Technically, it does not _have_ to be an import, but it seems like the only valid use case since (as you pointed out) the entire point of this package is to parse proto files.

Reply:
> Technically, it does not have to be an import

Hmmm, that seems confusing. As a user, I would expect that the filenames I provide would always be parsed from source. So I think it _should_ have to be an import, and we need a change in here so that we don't bother looking in here for the filenames provided. (If you aren't parsing source, it will likely be way faster to just use `desc.CreateFileDescriptor` to convert descriptor protos into rich cross-linked descriptors.)

Reply:
I changed the name to `DependencyDescriptors`.

I _can_ add a check, but I would rather not. It would add another argument to an already big function signature (in fairness, the function is unexported) to tell the difference between "the user asked for this file" vs. "this file was imported". (Also, you did not make a similar check on the well-known protos.)

I concur that it is unlikely that a user ever actually wants to do this, and it would be poor form to advertise that it works, but I expect that prohibiting it is needless. (Also, if someone _does_ have a reason to do it that somehow escapes both of us, it does not seem to be an issue if it just works.)

Reply:
Since we're looking at source files first, this seems fine. So let's go ahead and flesh out a test.

Reply:
Done.